burn-dataset/src/dataset/sqlite.rs

```diff
@@ -57,16 +63,48 @@ pub struct SqliteDataset<I> {
     columns: Vec<String>,
     len: usize,
     select_statement: String,
+    row_serialized: bool,
     phantom: PhantomData<I>,
 }
 
 impl<I> SqliteDataset<I> {
-    pub fn from_db_file(db_file: &str, split: &str) -> Self {
+    /// Initializes a `SqliteDataset` from a SQLite database file.
+    ///
+    /// This function sets up a `SqliteDataset` to read from a specific table in the SQLite database file.
+    /// The chosen table must share the same name as `split`, and must contain a primary key column named `row_id`.
+    ///
+    /// # Arguments
+    ///
+    /// * `db_file`: &str - The path to the SQLite database file.
+    ///
+    /// * `split`: &str - The name of the table in the database that the dataset is supposed to represent.
+    ///
+    /// * `row_serialized`: bool - Indicates the structure of the table. If `true`, each row in the table is expected
+    /// to have its fields MessagePack serialized into a single `item` column. If `false`, each field of the item
```

Comment:
... each row in the table is expected to be  __serialized with the MessagePack format__ in a single `item` column.